x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchFeature.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application;
+
+public class EnterpriseSearchFeature {
```

Comment:
I understand your instinct to make this strongly typed, but I have a couple of thoughts. 

First, I don't think we want the names of each individual API capitalized. So instead of `Search Application` I might suggest `search application` - this is more consistent with how we return error messages elsewhere as well as our documentation. 

Second, I'm not opposed to making this an enum, but I think it should be a top level enum and not part of an otherwise empty class. (See `QueryRuleCriteriaType` for an example of what I mean). Also could we come up with a more descriptive name than `EnterpriseSearchFeature`? I think `EnterpriseSearch` is redundant since we're already in the ent-search module. I suppose you could think about these as Products. You could also make a case for having static final Strings that we pass in as well.

Reply:
What about `public enum ProductType` instead of `enum Feature`?

Reply:
could we have the enum be defined as part of the `LicenseUtils`?
this enum is only used for licensing checks, so it would make sense to be part of `LicenseUtils`, instead of defining a standalone class.

Reply:
> You could also make a case for having static final Strings that we pass in as well.
@kderusso Can you please elaborate on this with an example? Do you mean to have static final Strings within the enum?

Reply:
Strings would replace an enum, however I like @ioanatia 's idea to move this to `LicenseUtils` - as a standalone class it's overkill.